#164055888 Implement feedback

deemukeni opened the pull request:
#### What does this PR do?

Implement challenge 3 feedback

#### Description of Task to be completed?

- Add comments and doc strings
- Get rid of redundant code

- Add validations on passwords and phone numbers

#### What are the relevant pivotal tracker stories?

[#164055888](https://www.pivotaltracker.com/story/show/164055888)

Commit:
[feature #164055888] Update README
Add comments and doc strings
Get rid of redundant code
Add validations on passwords and phone numbers

Commit:
[feature #164055888] Add comment

Commit:
[feature #164055888] Add comments

micahoriaso commented on tests/v2/test_endpoints.py:
```diff
@@ -96,20 +96,22 @@ def login(self):
+        sign_up_response = self.client.post("api/v2/auth/sign-up", data = json.dumps(self.user_signup_1), content_type='application/json')
         response = self.client.post("api/v2/auth/signin", data=json.dumps(self.user_login), content_type='application/json')
         result = json.loads(response.data.decode('utf-8'))
+        print (result)
```
Please remove this print statement

micahoriaso commented on api/v2/views/user_views.py:
```diff
@@ -17,6 +17,18 @@
 # User endpoints
 @bp.route("/auth/sign-up", methods=(["POST"]))
 def create_user():
+    """
```
Please refactor this. It should be of this format
```
"""
Blah blah blah
Blah blah blah
"""

micahoriaso commented on tests/v2/test_endpoints.py:
```diff
@@ -96,20 +96,22 @@ def login(self):
+        print (result)
+        result_2 = json.loads(sign_up_response.data.decode('utf-8'))
+
+        print (result_2)
```
Also remove this print statement

micahoriaso commented on api/v2/views/party_views.py:
```diff
@@ -90,6 +98,9 @@ def get_single_party(user, id):
 @bp.route("/parties/<int:id>", methods=["DELETE"])
 @token_required
 def delete_party(user, id):
+    """
```
Add params and returns to this docstring
```
"""
A method to delete a party
:param user: object, The user object
:param id: integer, Id of the party
:return: car details 
"""
```

micahoriaso commented on api/v2/views/party_views.py:
```diff
@@ -61,6 +63,9 @@ def create_party(user):
 @bp.route("/parties", methods=["GET"])
 @token_required
 def get_parties(user):
+    """
```
Update this docstring to the given specification

micahoriaso commented on api/v2/views/party_views.py:
```diff
@@ -75,6 +80,9 @@ def get_parties(user):
 @bp.route("/parties/<int:id>", methods=["GET"])
 @token_required
 def get_single_party(user, id):
+    """
```
Update to the given docstring specification

micahoriaso commented on api/v2/views/party_views.py:
```diff
@@ -75,6 +80,9 @@ def get_parties(user):
+    """
+    Get a party by its id
+    """
     party = Party.get_party_by_id(id)
```
Update to the given docstring specification

micahoriaso commented on api/v2/views/office_views.py:
```diff
@@ -69,6 +73,10 @@ def get_all_offices(user):
 @bp.route("/offices/<int:id>", methods=["GET"])
 @token_required
 def get_single_office(user, id):
+    """
```
Update to the given docstring specification

micahoriaso commented on api/v2/views/office_views.py:
```diff
@@ -83,6 +91,9 @@ def get_single_office(user, id):
 @bp.route("/offices/<int:id>", methods=(["DELETE"]))
 @token_required
 def delete_office(user, id):
+    """
```
Update to the given docstring specification

micahoriaso commented on api/v2/models/models.py:
```diff
@@ -254,14 +297,23 @@ def __init__(self, id, candidate_name, office_id, party_id):
         database.insert_to_db(query)
 
 
     def does_candidate_exist(candidate_id, office_id):
```
Update to the given docstring specification

micahoriaso commented on api/v2/models/models.py:
```diff
@@ -198,22 +240,23 @@ def get_user_by_phone_number(cls, phone_number):
 
     @classmethod
     def  get_user_by_email(cls, email):
+        """
```
Update to the given docstring specification

micahoriaso commented on api/v2/models/models.py:
```diff
@@ -198,22 +240,23 @@ def get_user_by_phone_number(cls, phone_number):
 
     @classmethod
     def  get_user_by_username(cls, username):
+        """
```
Update to the given docstring specification

Commit:
[feature #164055888]

Commit:
[feature #164055888]

Commit:
WIP: implement feedback.

- validate image url
- validate phone number field
- use appropriate status codes
